src/tikv/Region.cc

```diff
@@ -0,0 +1,274 @@
+#include <common/Exception.h>
+#include <tikv/Region.h>
+
+namespace pingcap
+{
+namespace kv
+{
+
+RPCContextPtr RegionCache::getRPCContext(Backoffer & bo, const RegionVerID & id, bool is_learner)
+{
+    RegionPtr region = getCachedRegion(bo, id);
+    if (region == nullptr)
+    {
+        throw Exception("not found region, region id is: " + std::to_string(id.id));
+    }
+    const auto & meta = region->meta;
+    auto peer = region->peer;
+
+    if (is_learner)
+    {
+        peer = region->learner;
```

Comment:
only support 1 learner per region?

Reply:
Indeed, I will remove the logic of Learner. ><
I will get learner address by config file, so client-c will support same functions with other clients.